backend/tests/integration/test_history_integration.py

```
@@ -1484,4 +1506,108 @@ def test_optional_metadata_persistence_and_retrieval(self, history_service_with_
+        
+        parent_execution_id = await history_service_with_test_db.create_stage_execution(parent_stage)
+        
+        # Create child stage executions with different statuses
+        child_agents = [
+            ("KubernetesAgent", StageStatus.COMPLETED, "Analysis complete"),
+            ("LogAgent", StageStatus.PAUSED, "Paused at iteration 2"),
+            ("NetworkAgent", StageStatus.FAILED, "Connection timeout"),
+        ]
+        
+        child_execution_ids = []
+        for idx, (agent_name, status, summary) in enumerate(child_agents):
```

Ruff (0.14.8)

Loop control variable `summary` not used within loop body

Rename unused `summary` to `_summary`

(B007)